filebeat/module/apache2/access/test/test.log-expected.json

```diff
@@ -1,205 +1,75 @@
 [
     {
-        "_index": "filebeat-2016.12.27",
-        "_type": "log",
-        "_id": "AVlBCaYsqYg9cc5KQfcT",
-        "_score": null,
-        "_source": {
-            "@timestamp": "2016-12-26T14:16:29.000Z",
-            "offset": 73,
-            "apache2": {
-                "access": {
-                    "response_code": "404",
-                    "remote_ip": "::1",
-                    "method": "GET",
-                    "user_name": "-",
-                    "http_version": "1.1",
-                    "body_sent": {
-                        "bytes": "209"
-                    },
-                    "url": "/favicon.ico"
-                }
-            },
-            "beat": {
-                "hostname": "192-168-0-7.rdsnet.ro",
-                "name": "192-168-0-7.rdsnet.ro",
-                "version": "6.0.0-alpha1"
-            },
-            "read_timestamp": "2016-12-27T15:52:23.304Z",
-            "source": "module/apache2/access/test/test.log",
-            "fields": {
-                "pipeline_id": "apache2-access-with_plugins",
-                "source_type": "apache2-access"
-            },
-            "prospector": {
-                "type": "log"
-            },
-            "input": {
-                "type": "log"
-            }
-        },
-        "fields": {
-            "@timestamp": [
-                1482761789000
-            ]
-        },
-        "sort": [
-            1482761789000
-        ]
-    },
+        "@timestamp": "2016-12-26T14:16:29.000Z", 
+        "apache2.access.body_sent.bytes": "209", 
+        "apache2.access.http_version": "1.1", 
+        "apache2.access.method": "GET", 
+        "apache2.access.remote_ip": "::1", 
+        "apache2.access.response_code": "404", 
+        "apache2.access.url": "/favicon.ico", 
+        "apache2.access.user_name": "-", 
+        "fileset.module": "apache2", 
+        "fileset.name": "access", 
+        "input.type": "log", 
+        "offset": 0, 
+        "prospector.type": "log"
```

Comment:
this is much more readable! I wonder if we should do this change for example `data.json` files, and Metricbeat tests in general

Reply:
My intention with this was actually to make it easier for machines.

I personally like the nested approach to read documents especially in the editor as I can "fold" parts. We could "flatten" all objects. What I worry is that people will think this is the way it's stored in Elasticsearch but it's actually stored as objects.